wit/keyvalue.wit

```diff
@@ -0,0 +1,29 @@
+/// a keyvalue interface
+resource keyvalue {
+	/// open a key-value store
+	static open: func(name: string) -> expected<keyvalue, keyvalue-error>
+
+	/// get the payload for a given key
+	get: func(key: string) -> expected<list<u8>, keyvalue-error> 
+
+	/// set the payload for a given key
+	set: func(key: string, value: list<u8>) -> expected<unit, keyvalue-error>
+
+	/// list the keys in the store
+	keys: func() -> expected<list<string>, keyvalue-error>
+
+	/// delete the payload for a given key
+	delete: func(key:string) -> expected<unit, keyvalue-error>
+}
+
+/// common keyvalue errors
+variant keyvalue-error {
+	key-not-found(string),
+	invalid-key(string),
+	invalid-value(string),
+	connection-error(string),
```

Comment:
> Error types like "connection-error", or "authentication-error" are related to how slight is connecting to the server-side.

This could indicate to the user that the authentication credentials they've provided are incorrect — I see value in surfacing these items. I think there still has to be more discussion over the `error` resource w/ the tracing function upstream before we adopt it in an implementation. For example — If that's all the error is, why is that a `kv-error`, and not just a generic error? Do we want a WASI capability for error handling?

Reply:
> If that's all the error is, why is that a kv-error, and not just a generic error? Do we want a WASI capability for error handling?

This is a good point. I see no reasons to not make it a generic error across SpiderLightning capabilities.

Reply:
How does that translate upstream? Currently, we've had individual error types in every proposal~ I figured that should be mirrored here.

Reply:
see: #278